.github/workflows/unused-deps.yml

```diff
@@ -0,0 +1,38 @@
+# Runs a series of sanity checks for crate consumers.
+
+name: sanity
```

Comment:
Nit: If we are only performing a `udep` check then the name could be more descriptive, e.g. `Unused dependency check`, as this name is load-bearing in the Actions overview. If there are plans to add more/consolidate with other checks then e.g. `Sanity checks` works.